### ISSUE TITLE
[Feature Request]: Custom groups for games library

### ISSUE THREAD

**Issue by EnigmaSigma** [PCSX2/pcsx2#11666]
### Description

It would be nice to be able to create custom group where you can put a series of games, or a genre, or a publisher in and then be able to open and close that group, similar how you can in the Steam Library.

I think a way to do this would be, in the view or the right click menu to have an option where you can create a new group, give it a name, and maybe an image.
Then to add the games to that group have a button within that group to add games where it comes up with the list of games and you can check them and then accept, or go to the individual game properties and be able to check a box for which groups it should appear in. That method would work for being able to remove games from the group too.
Then having an option to minimise and maximise the group to reduce all the games that are visible.

A future feature of this could be to have "smart groups" where you say create groups for genres and it searches all the games in your library and sorts them into genres (if games fit into multiple genres, it puts them in both). This could be done by looking at a resource like the gamesdb and filtering the genre section of each game.

### Reason

I like to name my dumped games as they are on the spine of the game which has an issue for a few games, specifically the Legacy of Kain series (Soul Reaver 2, Blood Omen 2 and Legacy of Kain - Defiance, that is how they are named on the spine). in my physical collection this is no issue as I can just put them in the right place, but in this library I can't group them together without changing their name.

### Examples

Steam's library in list view:
![image](https://github.com/user-attachments/assets/8c97ae13-25d2-4d72-846f-fd775bedac1c)

Steam's library in collection view:
![image](https://github.com/user-attachments/assets/c7cab1a0-c155-48b3-a4e9-bda1063ed3b7)

**Comment by OctopusButtons** [comment]
Yes another example for illustration is the "Collection" feature in OpenEmu on Mac, which uses the design structure of playlists from iTunes in the 2000's.

* In this example, picture attached, there's no programming for expanding the groups into a list of titles within the same list, it only shows the list of member items in the main app panel.  Highly effective even without the option to expand/collapse in the list panel itself. 
* User clicks File > New Collection, and can drag and drop games into a collection
  * or can also right-click on a game > pop-up menu > Add to Collection.

I think it would be a great organizational feature in PCSX2.  Mainly for getting a narrower view of a medium or large library, having groups/sets for easier browsing by mood or theme or any other user-defined point.

<img width="796" height="531" alt="Image" src="https://github.com/user-attachments/assets/0659e50a-53b9-4a45-9404-787c471c2d92" />

[EnigmaSigma](https://github.com/EnigmaSigma), note that the part of your suggestion about Smart Groups for automatic genre etc would I think require a whole database and programming for knowing and matching what the metadata is.
